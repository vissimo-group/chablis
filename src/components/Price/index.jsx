import styled, { css } from 'styled-components';

import {
  breakpoints,
  colors,
  fonts,
  fontSizes,
  fontWeights,
} from '../../styles/variables';

const Price = styled.p`
  position: relative;
  display: inline-block;
  font-family: ${fonts.DEFAULT};
  font-size: ${({ old }) => (old ? fontSizes.SMALL : fontSizes.BIG)};
  font-weight: ${({ old }) => (old ? fontWeights.MEDIUM : fontWeights.SEMIBOLD)};
  line-height: ${({ old }) => (old ? '18px' : '24px')};
  margin-top: 0;
  margin-bottom: 0;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;

  ${({ old, color }) => old && !color && css`
    color: ${colors.standard.medium_grey};
  `}

  ${({ old, color }) => old && color && css`
    color: ${color};
  `}

  ${({ old, color }) => !old && color && css`
    color: ${color || colors.standard.default_black};
  `}

  ${({ old }) => old && css`
    text-decoration-line: line-through;
  `}

  ${({ old }) => !old && css`
    @media screen and (min-width: ${breakpoints.SMALL}) {
      font-size: ${fontSizes.LARGE};
      line-height: 1;
    }

    small {
      display: inherit;
      font-size: ${fontSizes.SMALL};
      line-height: ${fontSizes.BIG};
      vertical-align: top;

      @media screen and (min-width: ${breakpoints.SMALL}) {
        font-size: ${fontSizes.BIG};
        line-height: 1;
      }
    }
  `}
`;

export default Price;
