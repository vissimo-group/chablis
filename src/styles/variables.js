const colors = {
  primary: {
    base: '#03a678',
    primary_20: '#088A66',
    primary_40: '#0D6F53',
    primary_60: '#125341',
  },

  secondary: {
    base: '#ce2a36',
    secondary_20: '#AA2731',
    secondary_40: '#87242C',
    secondary_60: '#632226',
  },

  tertiary: {
    base: '#FF9A22',
    tertiary_20: '#FFB966',
    tertiary_40: '#FFCA8C',
    tertiary_60: '#FFDCB3',
  },

  feedback: {
    info: '#2798E9',
    success: '#03CE61',
    warning: '#F5AA18',
    error: '#F45F5F',
  },

  neutrals: {
    neutral_100: '#1C1C1C',
    neutral_200: '#585757',
    neutral_300: '#969696',
    neutral_400: '#E7E7E7',
  },

  background: {
    primary: '#FFFFFF',
    secondary: '#F7F7F7',
  },

  standard: {
    orange_1: '#FFB55F',
    orange_2: '#F49A30',
    orange_3: '#F8892C',
    orange_4: '#EC720B',
    orange_5: '#C85F01',
    white: '#fff',
    lightest_grey: '#dcdcdc',
    light_grey: '#d3d3d3',
    medium_grey: '#c0c0c0',
    dark_grey: '#808080',
    darkest_grey: '#696969',
    darkest_grey_2: '#393939',
    black: '#20201e',
    default_black: '#000',
  },
};

const sizes = {
  DEFAULT: '16px',
  LARGE: '32px',
  HALF: '8px',
  RADIUS: '4px',
  FULL: '100%',
};

const breakpoints = {
  SMALL: '768px',
  NORMAL: '1024px',
  BIG: '1280px',
  LARGE: '1440px',
};

const fontSizes = {
  HUGE1: '48px',
  HUGE2: '36px',
  LARGE: '32px',
  BIG: '20px',
  NORMAL: '16px',
  SMALL: '14px',
  EXTRASMALL: '12px',
  TINY: '10px',
};

const fontWeights = {
  LIGHT: '300',
  REGULAR: '400',
  MEDIUM: '500',
  SEMIBOLD: '600',
  BOLD: '700',
};

export const zIndexes = {
  HELL: -1,
  GROUND: 1,
  SKY: 2,
  STRATOSPHERE: 3,
  MILKAWAY: 4,
  UNIVERSE: 5,
};

const fonts = {
  DEFAULT: 'Montserrat',
  BARLOW: 'Barlow Semi Condensed, sans-serif',
  PAYTONE: 'Paytone One, sans-serif', // TODO: Investigate issue with this font
};

export {
  breakpoints,
  colors,
  fonts,
  fontSizes,
  fontWeights,
  sizes,
};
