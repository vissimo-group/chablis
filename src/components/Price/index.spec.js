import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Price from '.';

import { colors } from '../../styles/variables';

describe('Price', () => {
  it('should render a price', () => {
    const element = (
      <Price>
        R$559
        <small>,90</small>
      </Price>
    );

    const tree = renderer.create(element).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render with specific text color', () => {
    const color = colors.standard.dark_grey;
    const element = <Price color={color}>R$559</Price>;

    const component = mount(element);
    expect(component.props().color).toEqual(color);
  });

  it('should render old Price', () => {
    const element = <Price old>855,90</Price>;
    const component = mount(element);
    expect(component.props().old).toEqual(true);
  });
});
