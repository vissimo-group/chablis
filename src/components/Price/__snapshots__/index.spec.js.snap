// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price should render a price 1`] = `
<p
  className="sc-aXZVg bBQjff"
>
  R$559
  <small>
    ,90
  </small>
</p>
`;
